spec/lib/bookingsync/engine/credentials_resolver_spec.rb

```diff
@@ -0,0 +1,18 @@
+require "spec_helper"
+
+RSpec.describe BookingSync::Engine::CredentialsResolver do
+  describe "#call" do
+    let(:credentials_resolver) do
+      BookingSync::Engine::CredentialsResolver.new("example.test")
+    end
+    let!(:application) do
+      Application.create(host: "example.test", client_id: "123", client_secret: "456")
+    end
+
+    it "returns an application credentials properly initiated" do
```

Comment:
it would be better to avoid a mock here and just do a proper setup or at least use a stub and do a dependency injection.